tests/unit/test_config.py

```diff
@@ -1007,6 +1008,32 @@ def test_rsync_includes_will_not_accept_sub_directories(tmp_flow_config):
     assert "Directories can only be from the top level" in str(exc.value)
 
 
+@pytest.mark.parametrize(
+    'cylc_var, expected_err',
+    [
+        ["CYLC_WORKFLOW_NAME", None],
+        ["CYLC_BEEF_WELLINGTON", (Jinja2Error, "is undefined")],
+    ]
+)
+def test_jinja2_cylc_vars(tmp_flow_config, cylc_var, expected_err):
+    """Valid CYLC variables should be available to Jinja2 during parsing."""
+    reg = 'nodule'
+    flow_file = tmp_flow_config(reg, """#!Jinja2
+    # {{""" + cylc_var + """}}
+    [scheduler]
+        allow implicit tasks = True
+    [scheduling]
+        [[graph]]
+            R1 = foo
+    """)
+    if expected_err is None:
+        WorkflowConfig(workflow=reg, fpath=flow_file, options=Values())
+    else:
+        with pytest.raises(expected_err[0]) as exc:
+            WorkflowConfig(workflow=reg, fpath=flow_file, options=Values())
+        assert expected_err[1] in str(exc)
```

Comment:
I think this is just a smidge faster.

```suggestion
    try:
        WorkflowConfig(workflow=reg, fpath=flow_file, options=Values())
    except expected_err[0] as exc:
        assert expected_err[1] in str(exc)
```

Reply:
@oliver-sanders  - I'm not care that much about this change - we can ignore it safely.

Reply:
(I've taken your suggestion)

Reply:
>    `except expected_err[0] as exc:`

Huh, flake8 no like this: *B030 Except handlers should only be names of exception classes*

Reply:
I don't like the `try...except` - we should use `pytest.raises()` to assert that a raise happens according to plan IMO

Reply:
Yeah, agreed. I'll revert.